src/ObjectUtils.js

```diff
@@ -81,6 +81,38 @@ var ObjectUtils = {
    */
   isPlainObject: function(obj) {
     return _.isObject(obj) && !_.isArray(obj);
+  },
+
+  /**
+   * Performs a deep clone of an object.
+   * @param {Object} source - the object to clone.
+   * @returns {Object} a clone of the given object, where any descendent objects are themselves
+   * cloned, and not just references to their respective counterpart in `source`.
+   */
+  deepClone: function(source) {
+    var clone = _.clone(source);
+    _.each(clone, (v,k) => {
+      if (_.isObject(v)) {
+        clone[k] = ObjectUtils.deepClone(v);
+      }
+    });
+    return clone;
+  },
+
+  /**
+   * Returns the changes between two objects. Includes attributes in `b` that are not in `a`, and
+   * attributes that are in both `a` and `b` but where the values differ. Performs deep comparison
+   * of objects and arrays.
```

Comment:
Can we clarify the semantics here around arrays? It's somewhat ambiguous at the moment if the code should return the full array when values change, or simple the values in the array that are changed (the first is the current behaviour as per this [test](https://github.com/mixmaxhq/backbone-publication/pull/11/files#diff-9eccd12b5bb0d9ee95122aeb932a308aR187)).